modules/navier_stokes/src/userobjects/RhieChowMassFlux.C

```diff
@@ -390,79 +404,87 @@ RhieChowMassFlux::populateCouplingFunctors(
   // We loop through the faces and populate the coupling fields (face H/A and 1/H)
   for (auto & fi : _fe_problem.mesh().faceInfo())
   {
-    Real face_rho = 0;
-    RealVectorValue face_hbya;
+    if (hasBlocks(fi->elemPtr()->subdomain_id()) ||
+        (fi->neighborPtr() && hasBlocks(fi->neighborPtr()->subdomain_id())))
```

Comment:
seems like we could use a dedicated faceInfo loop that takes a vector of subdomain ids (for later)

Reply:
Yes, this is a deficiency for sure. Unfortunately, I don't see an easy solution for this. The loop would also just add the same condition in a predicate for the iterator. The solution for this would be to cache the pointers to the faces and just iterate over those. I think that would not be too bad. Actually, I might just do that.